Fixed Laser Arrow

JMilamber opened the pull request.

JMilamber committed:
Fixed Laser Arrow

JMilamber committed:
remove un-needed accesstransformerentry, but leave the file in-case of future needs.

DerToaster98 committed:
restructure project

Move client stuff into client package
Move object-holders into init package

DerToaster98 committed:
move renderer registration to ClientEventSubscriber only

DerToaster98 committed:
make halberd able to block

Decided with brass to make it a weaponized shield basically...

TODO: Adjust models

DerToaster98 committed:
add in reach modifier for halberd

DerToaster98 committed:
depend on forge

DerToaster98 committed:
update implementation to fit forge

DerToaster98 committed:
add-in reach modifier for halberds

DerToaster98 committed:
TODO mark

DerToaster98 committed:
reach mod on dagger

DerToaster98 committed:
reach distance on spear

DerToaster98 committed:
add myself to authors

DerToaster98 committed:
well, i should implement the methods correctly, then it will work...

DerToaster98 committed:
don't just clear all other attributes...

FireController1847 commented on src/main/java/com/milamber_brass/brass_armory/BrassArmory.java:
```diff
@@ -25,27 +27,24 @@
     public static void onCommonSetup(FMLCommonSetupEvent event) {
         LOGGER.debug("Running common setup.");
-
+        
```
Is it possible to remove this whitespace?

FireController1847 commented on src/main/java/com/milamber_brass/brass_armory/BrassArmory.java:
```diff
@@ -25,27 +27,24 @@
-        // Register spear and arrow entity rendering handlers
-        RenderingRegistry.registerEntityRenderingHandler(BrassArmoryEntityTypes.SPEAR.get(), SpearEntityRenderer::new);
-        RenderingRegistry.registerEntityRenderingHandler(BrassArmoryEntityTypes.BA_ARROW.get(), BAArrowRenderer::new);
+        //Moved into ClientEventBusSubscriber
```
Let's remove the entire onClientSetup method from the main file if we're not going to put anything here. Presuming the ClientEventBusSubscriber will handle all of this anyways, no point in keeping this one.

FireController1847 replied:
Taking a closer look, let's just move the ClientEventBusSubscriber to the "client" package, then we can eliminate the "util" package entirely. Makes sense anyways, since it's client-handled stuff.

JMilamber replied:
Making changes and then removing this pull request until the weapons are actually ready to be merged.

JMilamber replied:
Done

FireController1847 commented on src/main/resources/META-INF/mods.toml:
```diff
@@ -20,3 +20,14 @@ Stays true to the core of Balkon's Weapons with new item textures made by Brass_
     versionRange="[1.16.5,1.17)"
     ordering="NONE"
     side="BOTH"
+[[dependencies.brass_armory]]	
```
This entire entry is not necessary since the loader version is already required at the top of the file, it doesn't need to be marked as a dependency (despite being shown in the Forge example)

JMilamber replied:
Oh, good to know lol

FireController1847 reviewed (changes requested):
Overall looks good, I haven't had a chance to test it in-game yet unfortunately but was able to take a first glance through the code. I left comments on specific things, here's some overall things before we merge this:

- I'm not sure I entirely agree with all of the formatting and refactoring changes being mixed in with the logic changes. I'd say if we can from here on out let's try and avoid refactors like this (myself included) without consulting the others and also putting it into a separate PR, otherwise we can't work parallel and ensuring branches stay up to date (like the quiver branch) would be virtually impossible since packages keep moving.

Most if not all logical changes seem to be good, however. Just a few minor other changes.

By the way, I'd check if your import-optimizer is disallowing the use of star imports. I noticed before I formatted some of the star imports from #3 were reverted. I don't care if we disallow them or not, but I do think we need to choose one otherwise this will keep happening each time we commit (it'll be changed, then reversed, then changed, and so on).

FireController1847 committed:
Run code through a formatter

JMilamber commented:
> * I'm not sure I entirely agree with all of the formatting and refactoring changes being mixed in with the logic changes. I'd say if we can from here on out let's try and avoid refactors like this (myself included) without consulting the others and also putting it into a separate PR, otherwise we can't work parallel and ensuring branches stay up to date (like the quiver branch) would be virtually impossible since packages keep moving.

My apologies, I gave Der Toaster the go-ahead to re-arrange stuff without thinking bout it, still adjusting to having so many people working on a project at once. 

These changes were also not meant to be added to this Pull Request, (this was supposed to be just for the laser arrow) but as the laser arrow was reworked in the melee branch all of Der Toasters' work got added to the pull request. 

Unsure if I should remove this pull request, or if we should try and commit this before further work is done, so that everyone can work off of the same file structure. 
@DerToaster98 @FireController1847 @Xratedjunior

FireController1847 commented:
While I was exploring the source the refactor seemed to be pretty good so I'd be okay with just including it in this PR. I don't think merging this PR will put any major stops or hurdles for other things regarding this project, so no worries. Honestly I probably wouldn't have even thought about it either - I didn't when I made PR 3 haha. You live you learn, right?

JMilamber committed:
Moved ClientEventBusSubcriber to client folder, removed util folder.

Removed onClientSetup from main file as it is no handled by ClientEventBusSibscriber

JMilamber committed:
removed extra dependency in mods.toml unless signifigant argument/reason for having it is supplied, as loaderversion is already given at top of file.

 According to Firecontroller it is not needed despite forge including it in their example

DerToaster98 committed:
don't change my access modifiers, thanks in advance

JMilamber commented on src/main/java/com/milamber_brass/brass_armory/entities/BAArrowEntity.java:
```diff
@@ -249,11 +256,12 @@ public void tick() {
-            // if the arrow enters an unloaded chunk, remove it.
+        // check that the arrow is not in the ground and has been flying for 1/5 a second.
+        else if (this.isArrowType(ArrowType.LASER) && !this.inGround && this.flightTime > 4) {
+            // call super.tick() twice to speed up arrow movement.
+            super.tick();
+            super.tick();
```
Doing so, at least in my testing,really messed with the block/entity hit detection. 

It might be better to remove the third call to super.tick(), but basically calling super.tick() on the arrow just runs all of the positional code a second time, thereby moving the arrow faster, as well as the hit detection code.

DerToaster98 replied:
That does not sound like a good solution, doing that really can mess up a lot cause some parts of the onUpdate could should only be called once per tick (well, the rest of the game and other mods  calculate with it only being called once per tick).

If you just want to change the speed, adjust the values in the constructor, iirc projectiles have a special constructor where the direction and velocity gets set.

JMilamber replied:
I'll look into that.
 for now, I'm going to merge this branch and look into that in a separate branch from the melee weapons

FireController1847 commented on src/main/java/com/milamber_brass/brass_armory/item/ICustomReachItem.java:
```diff
@@ -0,0 +1,54 @@
+public interface ICustomReachItem extends IForgeItem {
+
+    //Needs to be unique
+    public static final UUID REACH_DISTANCE_MODIFIER = UUID.fromString("b7d3df52-d360-491b-9bb5-2e8e3b5b279a");
```
These were changed by the formatter, apologies for removing them. No need to be passive aggressive with commit messages.

FireController1847 reviewed (approved):
lgtm

DerToaster98 commented:
What exactly do you mean by "formatter"? Do you just mean eclipse's built-in tool or a third party software? Cause for the first option i can provide the settings we use in cqr, that way all parties use the same formatting and it won't cause more issues.

FireController1847 commented:
> What exactly do you mean by "formatter"? Do you just mean eclipse's built-in tool or a third party software? Cause for the first option i can provide the settings we use in cqr, that way all parties use the same formatting and it won't cause more issues.

I'm using IntelliJ. Most notably I just right click on the package and hit "reformat code." Also I use the auto import optimizer. Not sure how to go about ensuring formatting stays the same between both programs

DerToaster98 commented:
It would be really appreciated if you could just adjust it so that it will just format code (e.g. editing line breaks and indentation(?)), i use auto import too but i'm not a fan of the star imports mentioned earlier (imo really just creates overhead when not everything from a package is used, also it decreases understandability and readability of the whole code)

So i think it should just work if the options for modifying the declarations are toggled off

FireController1847 commented:
> It would be really appreciated if you could just adjust it so that it will just format code (e.g. editing line breaks and indentation(?)), i use auto import too but i'm not a fan of the star imports mentioned earlier (imo really just creates overhead when not everything from a package is used, also it decreases understandability and readability of the whole code)
> 
> So i think it should just work if the options for modifying the declarations are toggled off

I'll check and see if I can do that. I've never had an issue with star imports due to compiler optimization, it only uses the needed packages the star import is simply to clean up the list of imports at the top of the list (realistically anymore, I understand that's not what it actually means but due to compiler optimization it doesn't require any more packages than necessary anyways, it just shortens the reference). Though I'm not sure I understand the argument against understandability and readability. How often are you actually looking at the imports for.. readability?

DerToaster98 commented:
On star imports: Yes, the compiler optimizes that. 
I look at the package way too often though. And normally i import packages using auto-import anyway

At the end of the day it is preference if one uses them or not, imo they just fall in the category of "syntax suggar" anyway.

I approve your suggestion, but we need something that keeps track of "who is working on what" (or i'm totally blind and don't see it)

FireController1847 commented:
@DerToaster98 Maybe we could utilize Javadocs @author annotation for classes? Then we can list authors for certain files. That could also be helpful if a specific file needs to be fixed or changed so the "owner" of that file can be consulted / notified.

DerToaster98 commented:
> 

Sounds good, we should define a class header then, used that in cqr when we were more than 2 developers and it helped indeed with identifying who is responsible. I'll search it after work and send it here, we might need to change it a bit though.

JMilamber commented:
I like this idea for author tracking a lot 👍. 
I have also created a new issue which contains who is working on which branches and a couple of rules, (basically what we've outlined here) so we are all on the same page and if anyone else joins they can be redirected there. 

If something needs to be worded better or changed or a rule added let me know on the issue.
